src/screens/InboundReceiveDetail/index.tsx

```diff
@@ -70,7 +70,7 @@ const InboundReceiveDetail = () => {
               shipmentItemId: shipmentItem.shipmentItemId,
               'container.id': shipmentItem['container.id'],
               'product.id': shipmentItem['product.id'],
-              'binLocation.id': shipmentItem['binLocation.id'],
+              'binLocation.id': state.receiveLocation,
```

Comment:
Why this helped? Looks like `state.receiveLocation` is equal to the `shipmentItem['binLocation.id']`.

Reply:
@awalkowiak Unless the user changes the receiving location in the UI and the state.receiveLocation is updated during that process.

What I don't like is that we're setting the state variable to the binLocation.name above. 
```
receiveLocation: shipmentItem['binLocation.name'],
```
It's basically a no-op since the user should be changing the receiving location in the UI before posting the request, but my assumption was that the state variables would hold the objects or object IDs and the components would have to work around that (i.e. if i wanted to display the binLocation.name i would reference the binLocation.name in the JSX). There are lots of concerning patterns around state management that we need to clean up.

cc @kumaransoftsuave